planner/core/exhaust_physical_plans.go

```diff
@@ -1558,6 +1558,27 @@ func (p *LogicalJoin) tryToGetIndexJoin(prop *property.PhysicalProperty) (indexJ
 	return append(allLeftOuterJoins, allRightOuterJoins...), false
 }
 
+func checkChildFitBC(p Plan) bool {
```

Comment:
it would alsow consider the size ration of two tables, for example, the data size of a small table is 10% of another large table.

Reply:
Hmmmm, I don't know If it will be better. Indeed, I'm trying to make code simpler now. More sophisticated optimization can be considered in the future. I will keep it in mind :D

Reply:
you can refer to other systems how to use broadcast join, such as MemSQL, GP and likewise systems.